tools/docker/nginx.apps.conf

```diff
@@ -1,14 +1,30 @@
 server {
-    listen       80;
-    listen  [::]:80;
-    server_name  localhost;
-    location / {
+        listen       80;
+        listen  [::]:80;
+        server_name  localhost;
+
         root   /usr/share/nginx/html;
-        index  index.html index.htm;
-        try_files $uri $uri/ /APP_NAME/index.html;
-    }
-    error_page   500 502 503 504  /50x.html;
-    location = /50x.html {
-        root   /usr/share/nginx/html;
-    }
+
+        server_tokens off;
+
+        location ~ /index.html|.*\.json$ {
+          expires -1;
+          add_header Cache-Control 'no-store, no-cache, must-revalidate, proxy-revalidate, max-age=0';
+        }
+
+        location ~ .*\.css$|.*\.js$ {
+          add_header Cache-Control 'max-age=31449600'; # one year
+        }
+
+        location / {
+          try_files $uri $uri/ /APP_NAME/index.html;
+
+          add_header Cache-Control 'max-age=2592000'; # 30 days
```

Comment:
doesn't that mean that the index.html file will be cached for a month, effectively preventing the browser from seeing new versions of the app?

Reply:
Also for all the other assets this might not be the best idea to cache them for so long; maybe one day would be better?

Reply:
generally, we should add hash to all static resources.
indeed this cannot be done for index.html, but at least for css and js, maybe for other assets as well.